tsdb/meta.go

```diff
@@ -796,9 +798,12 @@ func (m *Measurement) TagSets(shardID uint64, dimensions []string, condition inf
 		tags := make(map[string]string, len(dimensions))
 
 		// Build the TagSet for this series.
+		s.mu.RLock()
 		for _, dim := range dimensions {
-			tags[dim] = s.Tags.GetString(dim)
+			tags[dim] = s.tags.GetString(dim)
```

Comment:
Shouldn't this use `Tags()` now?  Could do it just before the for loop.  That way the `Series` is the only code responsible for locking it's `tags`.